server/rooms.ts

```diff
@@ -668,44 +668,42 @@ export abstract class BasicRoom {
 
 	/**
 	 * @param newID Add this param if the roomid is different from `toID(newTitle)`
+	 * @param moveLogs Whether or not to rename the log file. Defaults to true.
+	 * @param keepAliases Whether or not to point the room's aliases and name to its new name. Defaults to true.
 	 */
-	async rename(newTitle: string, newID?: RoomID) {
+	async rename(newTitle: string, newID?: RoomID, moveLogs = true, keepAliases = true) {
 		if (!newID) newID = toID(newTitle) as RoomID;
-		if (this.game || this.tour) return;
+		if ((this.type === 'battle' && moveLogs) || (this.type === 'chat' && this.game) || this.tour) return;
 
 		const oldID = this.roomid;
 		this.roomid = newID;
 		this.title = newTitle;
 		Rooms.rooms.delete(oldID);
-		Rooms.rooms.set(newID, this as ChatRoom);
+		Rooms.rooms.set(newID, this as Room);
 
 		if (oldID === 'lobby') {
 			Rooms.lobby = null;
 		} else if (newID === 'lobby') {
-			Rooms.lobby = this as ChatRoom;
+			Rooms.lobby = this as ChatRoom; // lobby shouldn't be a GameRoom
```

Comment:
I don't think this needs a comment?

```suggestion
			Rooms.lobby = this as ChatRoom;
```

A common rule of thumb is that a comment should say more than what the code already says. Like this is just repeating what `as ChatRoom` means.

Reply:
Heck, forget the cast, this might benefit from throwing if you try to rename a GameRoom to `lobby`.

Reply:
That's a good idea.